omod-common/src/main/java/org/openmrs/module/webservices/helper/TaskFactoryWrapper.java

```diff
@@ -0,0 +1,89 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.webservices.helper;
+
+import org.openmrs.scheduler.*;
+import org.openmrs.api.context.Context;
+import java.util.Collection;
+
+public class TaskFactoryWrapper {
+	
+	private SchedulerService schedulerService = Context.getSchedulerService();
+	
+	public TaskDefinition getTaskById(String id) {
+		return Context.getSchedulerService().getTask(Integer.parseInt(id));
+	}
+	
+	public Collection<TaskDefinition> getScheduledTasks() {
+		return schedulerService.getScheduledTasks();
+	}
+	
+	public Collection<TaskDefinition> getRegisteredTasks() {
+		return schedulerService.getRegisteredTasks();
+	}
+	
+	public boolean isTaskStarted(TaskDefinition task) {
+		return task.getStarted();
+	}
+	
+	public boolean isTaskStopped(TaskDefinition task) {
+		return !isTaskStarted(task);
+	}
+	
+	public void saveTaskDefinition(TaskDefinition task) {
+		schedulerService.saveTaskDefinition(task);
+	}
+	
+	public void unloadTask(TaskDefinition task) throws SchedulerException {
+		if (schedulerService.getTaskByName(task.getName()).getStarted()) {
+			schedulerService.shutdownTask(task);
+		}
+		schedulerService.deleteTask(task.getId());
+	}
+	
+	public void scheduleTask(TaskDefinition task) throws SchedulerException {
+		if (!schedulerService.getTaskByName(task.getName()).getStarted()) {
+			try {
+				schedulerService.scheduleTask(task);
+			}
+			catch (SchedulerException e) {
+				throw new UnsupportedOperationException();
```

Comment:
Does this really mean that the operation is not supported? The same applies elsewhere.